src/rai_toolbox/mushin/workflows.py

```diff
@@ -330,36 +403,184 @@ class MultiRunMetricsWorkflow(BaseWorkflow):
         loss        (epsilon, scale) float64 0.01 1.0 0.04 4.0 0.09 9.0
     """
 
+    # TODO: add target_job_dirs example
+    #      Document .swap_dims({"job_dir": <...>}) and .set_index(job_dir=[...]).unstack("job_dir")
+    #      for re-indexing based on overrides values
+
+    _JOBDIR_NAME: str = "job_dir"
+    _target_dir_multirun_overrides: Optional[DefaultDict[str, List[Any]]] = None
+    output_subdir: Optional[str] = None
+
+    # List of all the dirs that the multirun writes to; sorted by job-num
+    multirun_working_dirs: Optional[List[Path]] = None
+
     @abstractstaticmethod
-    def evaluation_task(*args: Any, **kwargs: Any) -> Dict[str, Any]:
-        raise NotImplementedError()
+    def evaluation_task(*args: Any, **kwargs: Any) -> Dict[str, Any]:  # type: ignore
+        """Abstract `staticmethod` for users to define the evalultion task"""
+
+    def run(
+        self,
+        *,
+        working_dir: Optional[str] = None,
+        sweeper: Optional[str] = None,
+        launcher: Optional[str] = None,
+        overrides: Optional[List[str]] = None,
+        version_base: Optional[Union[str, Type[_NotSet]]] = _NotSet,
+        target_job_dirs: Optional[Sequence[Union[str, Path]]] = None,
+        to_dictconfig: bool = False,
+        config_name: str = "rai_workflow",
+        job_name: str = "rai_workflow",
+        with_log_configuration: bool = True,
+        **workflow_overrides: Union[str, int, float, bool, dict, multirun, hydra_list],
+    ):
+        # TODO: add docs
+
+        if target_job_dirs is not None:
+            if isinstance(target_job_dirs, str):
+                raise TypeError(
+                    f"`target_job_dirs` must be a sequence of pathlike objects, got: {target_job_dirs}"
+                )
+            value_check("target_job_dirs", target_job_dirs, type_=Sequence)
+
+            target_job_dirs = [Path(s).resolve() for s in target_job_dirs]
+            for d in target_job_dirs:
+                if not d.is_dir() or not d.exists():
+                    # TODO: check not only that dir exists but that it contains
```

Comment:
This is challenging since we do not know what `hydra.output_subdir` is yet.  We could search `overrides` for it but there are probably other ways it can be set (e.g., yamls).   If we do `len(list(d.glob("**/config.yaml")))` do we assume the length is `==1` or `>=1`?   Do we allow multiple experiments to live in the `target_job_dir`?

Reply:
Yeah, we'll have to figure this out on a second pass; sticking with simple "does the directory exist" checks for now.